### DISCUSSION TITLE
UI Permissions Markers

### DISCUSSION THREAD

**Discussion by yaleman** [kanidm/kanidm#1066] [Ideas]
Should we add an extra type to UserAuthToken called "ui_hints" which is a BTreeSet of a new type "permission marker". These would be things like Oauth2, Groups, Persons, ServiceAccounts etc.  The idea is that when that ui_hint is set in the btreeset, we show that link / set of admin uis. Currently we can hardcode this to set "all the ui hints" in all accounts, but later we can actually make it derive from group membership to access controls so we can have the ui elements show up / hide as needed.

_Originally posted by @Firstyear in https://github.com/kanidm/kanidm/issues/1044#issuecomment-1251731223_

**Comment by Firstyear** [comment]
We've already started to add these, so isn't this an issue/dev request?

**Reply by yaleman** [reply]
Yep, #1116 applies
